includes/class-sensei-teacher.php

```diff
@@ -291,11 +291,13 @@ public function teacher_meta_box_content( $post ) {
 	 * @return array $users user id array
 	 */
 	public function get_teachers_and_authors() {
+		global $wp_version;
 
+		$authors_field     = version_compare( $wp_version, '5.9.0', '>=' ) ? 'capability' : 'who';
 		$author_query_args = array(
-			'blog_id' => $GLOBALS['blog_id'],
-			'fields'  => 'any',
-			'who'     => 'authors',
+			'blog_id'      => $GLOBALS['blog_id'],
+			'fields'       => 'any',
+			$authors_field => 'authors',
```

Comment:
I guess, to make the method more consistent for the different versions, you can try something like the following:

```php
$edit_course_roles = [];
foreach ( wp_roles()->roles as $role_slug => $role ) {
	if ( ! empty( $role['capabilities']['edit_course'] ) ) {
		$edit_course_roles[] = $role_slug;
	}
}

return get_users(
	[
		'blog_id'  => $GLOBALS['blog_id'],
		'fields'   => 'any',
		'role__in' => $edit_course_roles,
	]
);
```

This way you can avoid using the `who` and `capability` args and remove one query.

Reply:
@m1r0  I think this is a good approach. I will use the proposed solution, with a small change ```$role['capabilities']['edit_courses']``` instead of ```$role['capabilities']['edit_course']```

Also, I've written a test for this function to make sure I've not messed something up and to make sure it is working.

Reply:
Yep, you are right, sorry for the confusion.